pymongo/asynchronous/client_bulk.py

```diff
@@ -0,0 +1,765 @@
+            conn,
+            op_id,
+            listeners,  # type: ignore[arg-type]
+            None,
+            self.client.codec_options,
+        )
+
+        while self.idx_offset < self.total_ops:
+            # Construct the server command, specifying the relevant options.
+            cmd = {"bulkWrite": 1}
+            cmd["errorsOnly"] = not self.verbose_results
+            cmd["ordered"] = self.ordered  # type: ignore[assignment]
+            if self.bypass_doc_val is not None:
+                cmd["bypassDocumentValidation"] = self.bypass_doc_val
+            cmd["writeConcern"] = {"w": 0}  # type: ignore[assignment]
+            if self.comment:
+                cmd["comment"] = self.comment  # type: ignore[assignment]
+            if self.let:
+                cmd["let"] = self.let
+
+            conn.add_server_api(cmd)
+            ops = islice(self.ops, self.idx_offset, None)
+
+            # Run as many ops as possible in one server command.
+            to_send_ops, _ = await self._execute_batch_unack(bwc, cmd, ops)  # type: ignore[arg-type]
+
+            self.idx_offset += len(to_send_ops)
+
+    async def execute_command_unack_ordered(
+        self,
+        conn: AsyncConnection,
+    ) -> None:
+        """Execute commands with OP_MSG and w=0 WriteConcern, ordered."""
+        full_result: MutableMapping[str, Any] = {
+            "anySuccessful": False,
+            "error": None,
+            "writeErrors": [],
+            "writeConcernErrors": [],
+            "nInserted": 0,
+            "nUpserted": 0,
+            "nMatched": 0,
+            "nModified": 0,
+            "nDeleted": 0,
+            "insertResults": {},
+            "updateResults": {},
+            "deleteResults": {},
+        }
+        # Ordered bulk writes have to be acknowledged so that we stop
+        # processing at the first error, even when the application
+        # specified unacknowledged writeConcern.
+        initial_write_concern = WriteConcern()
+        op_id = _randint()
+        try:
+            await self._execute_command(
+                initial_write_concern,
+                None,
+                conn,
+                op_id,
+                False,
+                full_result,
+                self.write_concern,
+            )
+        except OperationFailure:
+            pass
+
+    async def execute_no_results(
+        self,
+        conn: AsyncConnection,
+    ) -> None:
+        """Execute all operations, returning no results (w=0)."""
+        if self.uses_collation:
+            raise ConfigurationError("Collation is unsupported for unacknowledged writes.")
+        if self.uses_array_filters:
+            raise ConfigurationError("arrayFilters is unsupported for unacknowledged writes.")
+        # Guard against unsupported unacknowledged writes.
+        unack = self.write_concern and not self.write_concern.acknowledged
+        if unack and self.uses_hint_delete and conn.max_wire_version < 9:
```

Comment:
I think we can remove these checks and instead add a top-level check for max_wire_version < 25 (MongoDB 8.0) and raise an error that it is not supported.